src/analysis_and_optimization/Pedantic_analysis.ml

```diff
@@ -75,12 +75,14 @@ let list_multi_twiddles (mir : Program.Typed.t) :
 let var_deps info_map label ?expr:(expr_opt : Expr.Typed.t option = None)
     (targets : string Set.Poly.t) : string Set.Poly.t =
   (* Labels of dependencies *)
-  let dep_labels =
+  (* let expr_vars_opt = Set.Poly.map ~f:fst (expr_var_set expr) in *)
+  let dep_labels, expr_vars =
     match expr_opt with
-    | None -> node_dependencies info_map label
+    | None -> (node_dependencies info_map label, Set.Poly.empty)
     | Some expr ->
-        let vars = Set.Poly.map ~f:fst (expr_var_set expr) in
-        node_vars_dependencies info_map vars label
+        let vvars = Set.Poly.map ~f:fst (expr_var_set expr) in
```

Comment:
It's a wrapper around string to represent variables, to make refactoring easier in case variables were ever not just strings. It's just used in the very oldest parts of code analysis. I might actually keep it around in order to handle indexed tuple expressions because they behavior like variables but aren't strings

Reply:
Honestly don't remember why it's called vvar, it might just have been to avoid colliding with var. I think it's just used in the code analysis sections and it should be replaced with lvalues when we can handle array indices. I just renamed it here to make it clearer that the type of this variable is vvar, nothing changed semantically